chain/cosmos/module_gov.go

```diff
@@ -52,6 +54,29 @@ func (tn *ChainNode) GovSubmitProposal(ctx context.Context, keyName string, prop
 
 // UpgradeProposal submits a software-upgrade governance proposal to the chain.
 func (tn *ChainNode) UpgradeProposal(ctx context.Context, keyName string, prop SoftwareUpgradeProposal) (string, error) {
+	if tn.IsAboveSDK47(ctx) {
+		cosmosChain := tn.Chain.(*CosmosChain)
+
+		authority, err := cosmosChain.GetGovernanceAddress(ctx)
```

Comment:
hmm, we can't really assume gov based authority with more PoA networks coming on.

`SoftwareUpgradeProposal` could add an authority string, but that may break json proto marshaling without having to write a custom decoder to ignore that key. or break the function signature to add a new authority string.

or leave it as is and fix it when it comes and issue or just override it on their side for ChainNode. Thoughts?

Reply:
i think i'm okay to put it in the `SoftwareUpgradeProposal` --does it get marshalled right now? I don't find anywhere that does so

Reply:
if not amazing that would work

Reply:
```go
		if prop.Authority == "" {
			authority, err := cosmosChain.GetGovernanceAddress(ctx)
			if err != nil {
				return "", err
			}
			prop.Authority = authority
		}
```

I tried to push but no perms. This works